gadgets/audit_seccomp/go/program.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"slices"
+
+	api "github.com/inspektor-gadget/inspektor-gadget/wasmapi/go"
+
+	ocispec "github.com/opencontainers/runtime-spec/specs-go"
+
+	seccomp "main/seccomp"
+)
+
+const (
+	// config.json is typically less than 100 KiB.
+	// 1 MiB should be enough.
+	configJsonMaxSize = 1 * 1024 * 1024
+)
+
+func updateConfig(configText string) string {
+	config := &ocispec.Spec{}
+	err := json.Unmarshal([]byte(configText), config)
+	if err != nil {
+		api.Errorf("unmarshaling config.json: %s", err)
+		return ""
+	}
+
+	if config.Linux == nil {
+		api.Errorf("Linux config is nil\n")
+		return ""
+	}
+	if config.Linux.Seccomp == nil {
+		api.Debugf("Seccomp config is nil. Creating one from containerd.\n")
+		sp := seccomp.DefaultProfile(config)
+		if !slices.Contains(sp.Flags, ocispec.LinuxSeccompFlagLog) {
+			sp.Flags = append(sp.Flags, ocispec.LinuxSeccompFlagLog)
+		}
+		config.Linux.Seccomp = sp
+	} else {
+		if slices.Contains(config.Linux.Seccomp.Flags, ocispec.LinuxSeccompFlagLog) {
+			api.Debugf("Seccomp config already has log flag\n")
+			return ""
+		}
+		config.Linux.Seccomp.Flags = append(config.Linux.Seccomp.Flags, ocispec.LinuxSeccompFlagLog)
```

Comment:
If the flag is not supported, maybe we can change the returned action to LOG? Or the default action or something?

Maybe a replace is enough? Like replace the action of kill*, errno, etc. with a log action, so we know which syscalls were hit and we can suggest a profile?

We will also be doing log for things that were log originally, but that is okay, we should include them in the audit analysis and suggest.

Reply:
> If the flag is not supported, maybe we can change the returned action to LOG? Or the default action or something

I don't think users expect this gadget to allow stuff that was previously blocked. Even if we put this on the documentation, users can miss it, so I'd prefer to avoid it.